### PR TITLE
update ref link for v1.16

### PR DESCRIPTION
**Pull request by kbhawkey**
Related to issue #18820.

### CODE REVIEW

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-18837--kubernetes-io-master-staging.netlify.com/docs/reference/

**Comment by kbhawkey** [comment]
@zacharysarah @jimangel 
Q: Should the `master` branch link to API reference snapshots in the reference overview page?
The reader can access the previous versions of the reference from the overview page and by accessing the snapshot from the release dropdown menu. Should each new docs release branch carry over (save a copy of) the previous versions of the generated API reference documentation?
<k8s-website>/static/docs/reference/generated/kubernetes-api/`<lots of versions here>`.

File: content/en/docs/reference/_index.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ This section of the Kubernetes documentation contains references.
 * [Kubernetes API Overview](/docs/reference/using-api/api-overview/) - Overview of the API for Kubernetes.
 * Kubernetes API Versions
   * [1.17](/docs/reference/generated/kubernetes-api/v1.17/)
-  * [1.16](/docs/reference/generated/kubernetes-api/v1.16/)
+  * [1.16](https://v1-16.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.16/)
   * [1.15](/docs/reference/generated/kubernetes-api/v1.15/)
   * [1.14](/docs/reference/generated/kubernetes-api/v1.14/)
   * [1.13](/docs/reference/generated/kubernetes-api/v1.13/)
```
Comment (sftim) [inline content/en/docs/reference/_index.md]: Does it make sense to make the same change for v1.13 to v1.15?

(I'm not sure what's best)

Comment (kbhawkey) [reply content/en/docs/reference/_index.md]: @sftim, Yes, I _think_ so. See my questions above ☝️
This would create changes during release time.

Comment (sftim) [reply content/en/docs/reference/_index.md]: I'm fine with either option so long as it makes sense to people.

**Comment by zacharysarah** [comment]
@kbhawkey Thanks for opening this!

>Should the master branch link to API reference snapshots in the reference overview page? The reader can access the previous versions of the reference from the overview page and by accessing the snapshot from the release dropdown menu. 

If the alternative is absolute links (aka this PR) then no, let's delete it for exactly the reasons you list.

> /static/docs/reference/generated/kubernetes-api/<lots of versions here>.

😳 

Oh geez. Yeah, let's observe our own CURRENT+4 policy for docs. Removing deprecated versions in `master` looks like a simple deletion PR, yes?

**Comment by kbhawkey** [comment]
@zacharysarah , looks like you are in a cleanup mode 😄 .
I did not raise the issue, but because the 1.16 version of the API reference has changed, the link on the API overview page points to the version of 1.16 on master.
The question is should the overview page continue to host links to the previous versions?
I can change this PR to remove the links (to previous versions) on the overview page.
The versions under  /static/docs/reference/generated/kubernetes-api/1.10 - 1.16 could be removed as well ( I have not fully thought this through).

**Comment by zacharysarah** [comment]
@kbhawkey You caught me. 😆 

> The question is should the overview page continue to host links to the previous versions?

No.

> I can change this PR to remove the links (to previous versions) on the overview page.

👍

**Comment by kbhawkey** [comment]
- Removed links to previous API refs and updated remaining link to use `latest-version`.
- Fixed other links (setup and cmd line tools).

**Pull request merged**